discord/app_commands/commands.py

```diff
@@ -2418,8 +2450,102 @@ async def my_guild_only_command(interaction: discord.Interaction) -> None:
     def inner(f: T) -> T:
         if isinstance(f, (Command, Group, ContextMenu)):
             f.guild_only = True
+            allowed_contexts = f.allowed_contexts or []
+            f.allowed_contexts = allowed_contexts
         else:
             f.__discord_app_commands_guild_only__ = True  # type: ignore # Runtime attribute assignment
+
+            allowed_contexts = getattr(f, '__discord_app_commands_contexts__', None) or []
+            f.__discord_app_commands_contexts__ = allowed_contexts  # type: ignore # Runtime attribute assignment
+
+        if AppCommandContext.GUILD not in allowed_contexts:
+            allowed_contexts.append(AppCommandContext.GUILD)
+
+        return f
+
+    # Check if called with parentheses or not
+    if func is None:
```

Comment:
I don't know if this is really needed because in other decorators that don't need extra parameters, library doesn't allow them to be used with parentheses (for example `@client.event`) so I think this would be inconsistent. The same goes for other two decorators.

Reply:
`@guild_only` has had this check before so I added them to the new decorators as well. Happy to remove it

Reply:
Right, my bad sorry 👍